includes/interwiki/Interwiki.php

```diff
@@ -295,6 +307,8 @@ protected static function getAllPrefixesDB( $local ) {
 		foreach ( $res as $row ) {
 			$retval[] = (array)$row;
 		}
+
+		$wgMemc->set( $key, $retval, $wgInterwikiExpiry ); // Wikia change
```

Comment:
Shouldn't we also check here if $retval is an Array? Few lines above we do set it to something which is not an Array (else in Memcache get) so if select fails we might get some unmeaningful value. Maybe it would be better not to use $retval as an value for cached data so we don't get it polluted in cache miss scenario?